multistreaming-server/rtmp-conf-generator.py

```diff
@@ -96,31 +113,50 @@ def generatePlatormPushURL(block_config):
 			).replace(
 				'%%REGION_CODE%%', region_code
 			)
+	elif block_config['platform'] == 'icecast':
+		push_url = block_config['icecastURL']
 	elif block_config['platform'] == 'custom':
 		push_url = block_config['customRTMPURL']
 
 
 	return push_url
 
 
+def getAudioTranscodeOptions(tc_conf):
+	return RTMP_TRANSCODE_AUDIO_OPTS_CUSTOM \
+		.replace('%%AUDIO_CODEC%%', tc_conf['audioCodec'] if 'audioCodec' in tc_conf else 'libfdk_aac') \
+		.replace('%%AUDIO_BIT_RATE%%', tc_conf['audioBitRate'] if 'audioBitRate' in tc_conf else '160k') \
+		.replace('%%AUDIO_SAMPLE_RATE%%', tc_conf['audioSampleRate'] if 'audioSampleRate' in tc_conf else '48000')
+
+
+def createIcecastApplicatonBlock(block_name, block_config):
```

Comment:
I was trying to avoid making destination specific block formats and instead destination specific formatting in the push URL, which is the only "touch point" with the destination. I was event trying to get the transcoding to be general and not formatted for a specific destination. Do you think that is possible here?